File: MountLicense/src/main/java/org/cubexmc/mountlicense/model/VehicleState.java
package org.cubexmc.mountlicense.model;

public enum VehicleState {
    ACTIVE,
    PARKED,
    LOCKED,
    RENTAL,
    MISSING,
    RELEASED;

    public static VehicleState fromString(String raw, VehicleState fallback) {
        if (raw == null) return fallback;
        try {
            return VehicleState.valueOf(raw.toUpperCase());
        } catch (IllegalArgumentException ex) {
            return fallback;
        }
    }
}

File: MountLicense/src/main/java/org/cubexmc/mountlicense/model/VehicleState.kt
package org.cubexmc.mountlicense.model

import java.util.Locale

enum class VehicleState {
    ACTIVE,
    PARKED,
    LOCKED,
    RENTAL,
    MISSING,
    RELEASED;

    companion object {
        @JvmStatic
        fun fromString(raw: String?, fallback: VehicleState): VehicleState {
            if (raw == null) {
                return fallback
            }
            return try {
                valueOf(raw.uppercase(Locale.getDefault()))
            } catch (ex: IllegalArgumentException) {
                fallback
            }
        }
    }
}
